lib/VM/JSLib/Intl.cpp

```diff
@@ -1424,6 +1452,14 @@ CallResult<HermesValue> intlDatePrototypeToSomeLocaleString(
     }
     toDateTimeOptions(*optionsRes, dtoFlags);
```

Comment:
It took me a while to understand what's going on here. These added checks are defined as part of [toDateTimeOptions](https://tc39.es/ecma402/#sec-todatetimeoptions), so please make them on that function. It also seems that steps 6., 7., and 8. are missing from the implementation (i.e., if either ```timeStyle``` or ```dateStyle``` are present, then ```needsDefault = false```).